src/services/pool/exchange/serializers/JoinParams.ts

```diff
@@ -54,15 +61,27 @@ export default class JoinParams {
     });
   }
 
-  private txData(amountsIn: BigNumberish[], minimumBPT: BigNumberish): string {
+  private txData(
+    amountsIn: BigNumberish[],
+    minimumBPT: BigNumberish,
+    isProportional: boolean
+  ): string {
     if (this.exchange.pool.onchain.totalSupply === '0') {
       return this.dataEncodeFn({ kind: 'Init', amountsIn });
     } else {
-      return this.dataEncodeFn({
-        kind: 'ExactTokensInForBPTOut',
-        amountsIn,
-        minimumBPT
-      });
+      // Investment Pools can only be joined proportionally if trading is hallted
+      if (this.isInvestmentPool && isProportional && !this.isSwapEnabled) {
```

Comment:
@EndymionJkb I don't think the `isProportional` is required here. We are hiding custom inputs in the form if it's an investment pool in this state so you can assume it is proportional.

Reply:
Yes - I wrote that before I hid the custom inputs